File: frontend/src/api/audit.ts
/** This backend has no separate audit-log table — the investigation
 * timeline IS its audit trail (see docs/reporting/reports.md's "Audit
 * report" section, an established precedent since Phase 9/11/12/13).
 * Proposed REST contract: GET /api/v1/audit — a target-wide read across
 * every investigation's timeline, exactly what `ai-recon report create
 * --type audit` already does server-side. */
import { USE_MOCKS } from "./config";
import { apiRequest } from "./client";
import { mockPage } from "@/mocks/paginate";
import { timelineEvents as mockTimeline } from "@/mocks/database";
import type { TimelineEvent } from "@/types/investigation";
import type { Page, PageParams } from "@/types/common";

export async function listAuditEvents(targetId: string, page?: PageParams): Promise<Page<TimelineEvent>> {
  if (USE_MOCKS) {
    const items = [...mockTimeline.filter((t) => t.targetId === targetId)].sort((a, b) =>
      b.timestamp.localeCompare(a.timestamp),
    );
    return mockPage(items, page);
  }
  return apiRequest<Page<TimelineEvent>>("/api/v1/audit", { searchParams: { target_id: targetId, limit: page?.limit, cursor: page?.cursor } });
}

File: internal/api/audit.go
package api

// This backend has no separate audit-log table — the investigation
// timeline IS its audit trail (see docs/reporting/reports.md's own
// "Audit report" precedent, established since Phase 9/11/12/13).
// Mirrors frontend/src/api/audit.ts: a target-wide read across every
// investigation's timeline.

import (
	"net/http"

	investigationrepo "ai-recon-platform/internal/repository/investigation"
)

func (h *handler) listAuditEvents(w http.ResponseWriter, r *http.Request) {
	targetID, ok := requiredTargetID(w, r)
	if !ok {
		return
	}
	invPage, err := h.deps.Investigation.List(r.Context(), investigationrepo.ListFilter{TargetID: targetID, Pagination: pagination500()})
	if err != nil {
		writeError(w, h.logger, err)
		return
	}

	var out []timelineEventDTO
	for _, inv := range invPage.Items {
		page, err := h.deps.Investigation.Timeline(r.Context(), inv.ID, true, pagination500())
		if err != nil {
			continue
		}
		for _, e := range page.Items {
			out = append(out, toTimelineEventDTO(e))
		}
	}
	if out == nil {
		out = []timelineEventDTO{}
	}
	writeJSON(w, http.StatusOK, pageResponse[timelineEventDTO]{Items: out, HasMore: false})
}
